### PR TITLE
Hide cursor and fix dots

### PR DESCRIPTION
**Pull request by daniel-weisse**
### Proposed change(s)

- Fixes dots not being displayed correctly and hides the cursor while printing them

### CODE REVIEW

**Review by datosh — APPROVED** [review]
LGTM

File: cli/internal/cmd/spinner.go [hunk 1]
```diff
@@ -8,39 +8,46 @@ package cmd
...
 
-	"github.com/spf13/cobra"
+const (
+	// hideCursor and showCursor are ANSI escape sequences to hide and show the cursor.
+	hideCursor = "\033[?25l"
+	showCursor = "\033[?25h"
 )
 
 var (
 	spinnerStates = []string{"⣾", "⣽", "⣻", "⢿", "⡿", "⣟", "⣯", "⣷"}
```
Comment (datosh) [inline cli/internal/cmd/spinner.go:24]:
```suggestion
	spinnerStates = []string{"⣷", "⣯", "⣟", "⡿", "⢿", "⣻", "⣽", "⣾"}
```

**Comment by daniel-weisse** [comment]
Made some more changes:
* Spinner can be restarted
* Try to detect if out is a tty, if not, restrict messages to once every 20 seconds
  * @Nirusu could you test if this works for darwin?
* Add the `Writer` method to `spinner` directly instead of holding it in a wrapped type

**Comment by katexochen** [comment]
Did you check this works on interrupt? Code lgtm.

**Comment by daniel-weisse** [comment]
>Did you check this works on interrupt?

Works for interrupts, but I just noticed some issues with the slowed down spinner.
Will refactor slightly

**Comment by Nirusu** [comment]
I won’t have time to check this for a couple of days, but you could use the e2e action and check if it disables / slows down the spinner.

It should likely be fine though, the check should generally work for UNIX and the go-isatty package seems to be multi OS aware.

**Review by Nirusu — APPROVED** [review]
TTY tested on fish and non-TTY tested on CI, both on macOS.
I hope that covers weird edge cases :D

Personally I think we could append "..." to the non-TTY input just for clarity that this is a longer-waiting process, but I don't consider non-TTY to be super important so decide by yourself.

**Pull request merged**